code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -0,0 +1,59 @@
+/* global React, dashboard */
+
+window.dashboard = window.dashboard || {};
+
+window.dashboard.STAGE_PROGRESS_TYPE = React.PropTypes.arrayOf(React.PropTypes.shape({
+  title: React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.number]),
+  status: React.PropTypes.string,
+  kind: React.PropTypes.oneOf(['unplugged', 'assessment', 'puzzle']),
+  link: React.PropTypes.string,
+  id: React.PropTypes.number
+}));
```

Comment:
This feels a little weird (sharing PropTypes across components). I think a big part of that weirdness comes from the fact that we're sticking it on window.dashboard, which we don't need to do anymore (we can just require things :)). However, even in that world I'm not sure of the best way to share this. My first thought is to stick it in a third file that gets required by both components, but that also feels a little strange.